src/views/view.c

```diff
@@ -1222,6 +1226,85 @@ int dt_view_image_expose(dt_view_image_over_t *image_over, uint32_t imgid, cairo
   return missing;
 }
 
+void
```

Comment:
Well, at lest this way that code is not duplicated. This looks better, at least to me.